client/client.go

```diff
@@ -69,6 +71,44 @@ type ApiRequester interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
+// APIError wraps an error with the request_id - and if
+// the server returns an error message, this is also returned.
+type APIError struct {
+	error
+	reqID        string
+	serverErrMsg string
+}
+
+func NewAPIError(err error, resp *http.Response) *APIError {
+	a := APIError{
+		error: err,
+		reqID: resp.Header.Get("request_id"),
+	}
+
+	if resp.StatusCode >= 400 && resp.StatusCode < 600 {
+		a.serverErrMsg = unmarshalErrorMessage(resp.Body)
+	}
+	return &a
+}
+
+func (a *APIError) Error() string {
+
+	err := fmt.Sprintf("(request_id: %s): %s", a.reqID, a.error.Error())
```

Comment:
Can we move this inside `if a.serverErrMsg != ""` below? There is no need to evaluate it if there is no error as we are returning an empty string in this case anyway.

Reply:
changed it to return the original error: ``` 
func (a *APIError) Error() string {

	err := fmt.Sprintf("(request_id: %s): %s", a.reqID, a.error.Error())

	if a.serverErrMsg != "" {
		return err + fmt.Sprintf(" server error message: %s", a.serverErrMsg)
	}

	return err

}
```
Which it was supposed to all along.

Reply:
it is simply a server-message append.